### PR TITLE
a11y additions

### PR DESCRIPTION
**Pull request by mxmason**
(Based on #6; reopened for better git history)

In this pr:
- No more modal! 🎉 
  - The `What` link takes users down to an info section that is already in the DOM. The end.
- A `hear state of game` button which generates an aural update for screen-reader users
- Better DOM structure, for a11y reasons as well as just-because reasons
- Functional components where possible
- Bigger text and a boxy style
- A major version bump to all our dependencies

Thanks again to @jpdevries for kicking off the spiritual predecessor of these changes ages ago!

### CODE REVIEW

File: src/components/feedback.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+import './feedback.css';
+
+export default function Feedback(props) {
+  /** 
+   * use the guessCount to generate a key key so that React treats the feedback message
```
Comment (oampo) [inline src/components/feedback.js]: Duplicate word: `key`

File: src/components/guess-count.js [hunk 2]
```diff
@@ -3,9 +3,12 @@ import React from 'react';
...
-        <p>
-            Guess #<span id="count">{props.count}</span>!
-        </p>
-    );
+  const isPlural = props.guessCount !== 1;
+  const guessNoun = isPlural ? 'guesses' : 'guess';
+
+  return (
+    <h2 id="guessCount">
+      You&apos;ve made <span id="count">{props.guessCount}</span> {guessNoun}!
```
Comment (oampo) [inline src/components/guess-count.js]: Why do we need &apos; here?  Can we not just use an actual apostrophe?

File: src/components/guess-list.js [hunk 3]
```diff
@@ -3,15 +3,15 @@ import React from 'react';
 import './guess-list.css';
 
 export default function GuessList(props) {
-    const guesses = props.guesses.map((guess, index) => (
-        <li key={index}>
-            {guess}
-        </li>
-    ));
+  const guessesHTML = props.guesses.map((guess, index) => (
```
Comment (oampo) [inline src/components/guess-list.js]: Why the change of name to `guessesHTML`?  It's not even HTML :)

File: src/index.css [hunk 4]
```diff
@@ -4,48 +4,127 @@
...
 .clearfix:after {
-    clear: both;
+  clear: both;
 }
 
 * {
-	-webkit-box-sizing: border-box;
-	-moz-box-sizing: border-box;
-	box-sizing: border-box;
+  -webkit-box-sizing: border-box;
```
Comment (oampo) [inline src/index.css]: Create React App uses an autoprefixer, so we can can the `-webkit` and `-moz` prefixes

**Review by oampo — APPROVED** [review]
Couple of tiny things, then ready to merge

File: src/components/guess-count.js (hunk 2 above)
Comment (mxmason) [reply src/components/guess-count.js]: Unless v16 got less picky, the app warns about inescaped entity characters placed in the render function.
